app/ai_backend/genesis_evolutionary_conduit.py

```
@@ -162,1760 +134,63 @@ def __init__(self):
+        self.evolution_active = False
+        print("💤 EvolutionaryConduit shutting down")
 
-# Wait for analysis threads to complete
-for thread_name, thread in self.analysis_threads.items():
-    if thread.is_alive():
-        thread.join(timeout=2.0)
+    def _generate_insight_id(self, base_name: str) -> str:
+        """Generate unique insight ID"""
+        timestamp = str(int(time.time() * 1000))
+        content = f"{base_name}_{timestamp}"
+        return hashlib.md5(content.encode()).hexdigest()[:12]
```

Ruff (0.14.1)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)